server/src/main/java/org/opensearch/action/admin/cluster/remotestore/RemoteStoreService.java

```diff
@@ -0,0 +1,160 @@
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.RepositoriesMetadata;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.repositories.Repository;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Supplier;
+
+/**
+ * Contains all the method needed for a remote store backed node lifecycle.
+ */
+public class RemoteStoreService {
+
+    private static final Logger logger = LogManager.getLogger(RemoteStoreService.class);
+    private final Supplier<RepositoriesService> repositoriesService;
+    private final ThreadPool threadPool;
+    public static final Setting<CompatibilityMode> REMOTE_STORE_COMPATIBILITY_MODE_SETTING = new Setting<>(
+        "remote_store.compatibility_mode",
+        CompatibilityMode.STRICT.name(),
+        CompatibilityMode::parseString,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    /**
+     * Node join compatibility mode introduced with remote backed storage.
+     *
+     * @opensearch.internal
+     */
+    public enum CompatibilityMode {
+        STRICT("strict"),
+        ALLOW_MIX("allow_mix");
+
+        public final String mode;
+
+        CompatibilityMode(String mode) {
+            this.mode = mode;
+        }
+
+        public static CompatibilityMode parseString(String compatibilityMode) {
+            try {
+                return CompatibilityMode.valueOf(compatibilityMode.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(
+                    "["
+                        + compatibilityMode
+                        + "] compatibility mode is not supported. "
+                        + "supported modes are ["
+                        + CompatibilityMode.values().toString()
+                        + "]"
+                );
+            }
+        }
+    }
+
+    public RemoteStoreService(Supplier<RepositoriesService> repositoriesService, ThreadPool threadPool) {
+        this.repositoriesService = repositoriesService;
+        this.threadPool = threadPool;
+    }
+
+    /**
+     * Performs repository verification during node startup post its creation by invoking verify method against
+     * repository mentioned. This verification will happen on a local node to validate if the node is able to connect
+     * to the repository.
+     */
+    public void verifyRepositoriesLocally(List<Repository> repositories, DiscoveryNode localNode) {
```

Comment:
Lets merge createRepo and verifyRepo to ensure creation is followed by verification explicitly